pkg/network/address.go

```diff
@@ -83,6 +83,13 @@ func multiaddrStringFromHostAddr(host string) (string, error) {
 		return "", err
 	}
 
+	// Empty address in host :8080` generates `/dns4//tcp/8080` multiaddr
+	// which is invalid. It could be `/tcp/8080` but this breaks
+	// `manet.DialArgs`. The solution is to manually parse it as 0.0.0.0
+	if endpoint == "" {
+		endpoint = "0.0.0.0"
```

Comment:
We can return from here immediately, without further parsing.

Reply:
Yeah, fixed, thanks.